handlers/people_test.go

```diff
@@ -258,3 +258,191 @@ func TestDeletePerson(t *testing.T) {
+		handler.ServeHTTP(rr, req)
+
+		var returnedPeople []db.Person
+		err = json.Unmarshal(rr.Body.Bytes(), &returnedPeople)
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusOK, rr.Code)
+		assert.EqualValues(t, expectedPeople, returnedPeople)
+		mockDb.AssertExpectations(t)
+	})
+}
+
+func TestGetListedPeople(t *testing.T) {
+	mockDb := mocks.NewDatabase(t)
+	pHandler := NewPeopleHandler(mockDb)
+
+	t.Run("should return all listed users", func(t *testing.T) {
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetListedPeople)
+		expectedPeople := []db.Person{
+			{ID: 1, Uuid: "uuid1", OwnerPubKey: "pubkey1", OwnerAlias: "John Doe"},
+			{ID: 2, Uuid: "uuid2", OwnerPubKey: "pubkey2", OwnerAlias: "John Smith"},
+		}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("page", "1")
+		rctx.URLParams.Add("limit", "10")
+		req, err := http.NewRequestWithContext(context.WithValue(context.Background(), chi.RouteCtxKey, rctx), http.MethodGet, "/", nil)
+		assert.NoError(t, err)
+
+		mockDb.On("GetListedPeople", mock.Anything).Return(expectedPeople)
+		handler.ServeHTTP(rr, req)
+
+		var returnedPeople []db.Person
+		err = json.Unmarshal(rr.Body.Bytes(), &returnedPeople)
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusOK, rr.Code)
+		assert.EqualValues(t, expectedPeople, returnedPeople)
+		mockDb.AssertExpectations(t)
+	})
+
+	t.Run("should return only users that match a search text when a search is added to the URL query", func(t *testing.T) {
+		mockDb.ExpectedCalls = nil
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetListedPeople)
+		expectedPeople := []db.Person{
+			{ID: 1, Uuid: "uuid1", OwnerPubKey: "pubkey1", OwnerAlias: "John Doe"},
+			{ID: 2, Uuid: "uuid2", OwnerPubKey: "pubkey2", OwnerAlias: "John Smith"},
+		}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("page", "1")
+		rctx.URLParams.Add("limit", "10")
+		rctx.URLParams.Add("search", "John")
+		req, err := http.NewRequestWithContext(context.WithValue(context.Background(), chi.RouteCtxKey, rctx), http.MethodGet, "/", nil)
+		assert.NoError(t, err)
+
+		mockDb.On("GetListedPeople", mock.Anything).Return(expectedPeople)
+		handler.ServeHTTP(rr, req)
+
+		var returnedPeople []db.Person
+		err = json.Unmarshal(rr.Body.Bytes(), &returnedPeople)
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusOK, rr.Code)
+		assert.EqualValues(t, expectedPeople, returnedPeople)
+		mockDb.AssertExpectations(t)
+	})
+
+	t.Run("should return only users that match a skill set when languages are passed to the URL query", func(t *testing.T) {
+		mockDb.ExpectedCalls = nil
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetListedPeople)
+		expectedPeople := []db.Person{
+			{ID: 1, Uuid: "uuid1", OwnerPubKey: "pubkey1", OwnerAlias: "John Doe"},
+		}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("page", "1")
+		rctx.URLParams.Add("limit", "10")
+		rctx.URLParams.Add("languages", "typescript")
```

Comment:
@AbdulWahab3181 the page, limit, and languages should be in the URL query, not the param.

Reply:
Addressed